CONTRIBUTING.md

```diff
@@ -0,0 +1,206 @@
+
+**How to Submit A Good Bug Report**
+
+Bugs are tracked as [GitHub issues](https://guides.github.com/features/issues/). After figuring out the issue, create an issue on that repository and provide all the information, screen shot also if necessary.
+
+Explain the problem and include additional details to help maintainers reproduce the problem:
+
+- **Use a clear and descriptive title for the issue** to identify the problem.
+- **Describe the exact steps which reproduce the problem** in as many details as possible.
+- **Provide specific examples to demonstrate the steps.** Include links to files or GitHub projects, or copy/pasteable snippets, which you use in those examples.
+- **Describe the behavior you observed** after following the steps and point out what exactly is the problem with that behavior.
+- **Explain which behavior you expected** to see instead and why.
+- **Include screenshots and animated GIFs** which show you following the described steps and clearly demonstrate the problem.
+- **If the problem wasn't triggered by a specific action, describe what you were doing before the problem happened** and share more information using the guidelines below.
+
+### **Suggesting Enhancements ✨**
+
+This section guides you through submitting an enhancement suggestion for OurPromise, including completely new features and minor improvements to existing functionality. Following these guidelines helps maintainers and the community understand your suggestion and find related suggestions 🔎.
+
+Before creating enhancement suggessions, please check if its already existing or not. When you are creating an enhancement suggestion, please include as many details as possible. Just include the steps that you imagine you would take if the feature you're requesting existed.
+
+**Before Submitting An Enhancement Suggestion**
+
+- You might discover that the enhancement is already available.
+- Check if there's a package which provides the same.
+- Check if it was already requested.
+
+**How To Submit A Good Enhancement Suggestion?**
+
+Enhancement suggestions are tracked as [GitHub issues](https://guides.github.com/features/issues/). After deciding your enhancement create an issue in the issues section.
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+- **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+- **Describe the current behavior and explain which behavior you expected to see instead** and why.
+- **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of the project which the suggestion is related to.
+- **Explain why this enhancement would be useful** to our users.
```

Comment:
[Note]
this is good for now, but we will need an issue template for this
The issue opened at #51